api/src/database/database.py

```diff
@@ -89,16 +90,21 @@ def start_new_db_session(self):
         finally:
             lock.release()
 
+    def should_close_db_session(self):
+        return os.getenv("%s" % SHOULD_CLOSE_DB_SESSION, "false").lower() == "true"
+
     def close_session(self):
         """
         Closes a session
         :return: True if the session was started, False otherwise
         """
         try:
-            if global_session is not None and global_session.is_active:
+            should_close = self.should_close_db_session()
+            if should_close == "true" and global_session is not None and global_session.is_active:
```

Comment:
As `should_close_db_session` returns a boolean, this conditional should be:
```suggestion
            if should_close and global_session is not None and global_session.is_active:
```